Twispay WooCommerce Plugin/twispay/includes/twispay-processor.php

```diff
@@ -21,236 +23,151 @@
-        $inputs['identifier'] = '_' . $data['customer_id'];
-        
-        // InvoiceEmail
-        $inputs['invoiceEmail'] = '';
-        
-        // Item names
-        $items = $order->get_items();
-        $i = 0;
-        foreach ( $items as $item ) {
-            $inputs['item[' . $i . ']'] = $item['name'];
-            $inputs['units[' . $i . ']'] = $item['quantity'];
-            $inputs['subTotal[' . $i . ']'] = number_format( ( float )$item['subtotal'], 2 );
-            $inputs['unitPrice[' . $i . ']'] = number_format( number_format( ( float )$item['subtotal'], 2) / number_format( ( float )$item['quantity'], 2 ), 2 );
-            
-            $i += 1;
-        }
-        
-        // Lastname
-        $inputs['lastName'] = '';
-        $inputs['lastName'] = $data['billing']['last_name'];
-        if ( ! $inputs['lastName'] ) {
-            $inputs['lastName'] = $data['shipping']['last_name'];
-        }
-        
-        // Order ID
-        $inputs['orderId'] = '';
-        if ( isset( $_GET['tw_reload'] ) && $_GET['tw_reload'] ) {
-            $inputs['orderId'] = $data['id'] . '_' . time();
-        }
-        else {
-            $inputs['orderId'] = $data['id'];
-        }
-        
-        // Order tags
-        $inputs['orderTags[0]'] = '';
-        
-        // Order type
-        $inputs['orderType'] = 'purchase';
-        
-        // Site ID
-        $inputs['siteId'] = $siteID;
-        
-        // Postcode
-        $inputs['zipCode'] = $data['billing']['postcode'];
-        if ( ! $inputs['zipCode'] ) {
-            $inputs['zipCode'] = $data['shipping']['postcode'];
+
+        /* Extract the customer details. */
+        $customer = [ 'identifier' => (0 == $data['customer_id']) ? ('_0_' . rand()) : ('_' . $data['customer_id'])
+                    , 'firstName' => ($data['billing']['first_name']) ? ($data['billing']['first_name']) : ($data['shipping']['first_name'])
+                    , 'lastName' => ($data['billing']['last_name']) ? ($data['billing']['last_name']) : ($data['shipping']['last_name'])
+                    , 'country' => ($data['billing']['country']) ? ($data['billing']['country']) : ($data['shipping']['country'])
+                    /* , 'state' => ($data['billing']['state']) ? ($data['billing']['country']) : ($data['shipping']['country']) */
+                    , 'city' => ($data['billing']['city']) ? ($data['billing']['city']) : ($data['shipping']['city'])
+                    , 'address' => ($data['billing']['address_1']) ? ($data['billing']['address_1']/* . ' ' . $data['billing']['address_2']*/) : ($data['shipping']['address_1']/* . ' ' . $data['shipping']['address_2']*/)
+                    , 'zipCode' => ($data['billing']['postcode']) ? ($data['billing']['postcode']) : ($data['shipping']['postcode'])
+                    /* , 'phone' => $data['billing']['phone'] */
+                    , 'email' => $data['billing']['email']
+                    , 'tags' => []

```

Comment:
We should comment this line as `tags` is not a required field.

Reply:
Done